### PR TITLE
travis and makefile cleanup

### PR DESCRIPTION
**Pull request by stephengroat**
make xenial default in travis
remove separate cmocka script
add android builds
move fuzz testing to base makefile

### CODE REVIEW

**Comment by aquynh** [comment]
is this ready to merge?

**Comment by stephengroat** [comment]
@aquynh should be good to go. gets a bunch of cleanup in and improves test coverage

File: .travis.yml [hunk 1]
```diff
@@ -1,56 +1,75 @@
...
-before_install:
-        - export LD_LIBRARY_PATH=`pwd`/tests/:$LD_LIBRARY_PATH
 before_script:
-        - wget https://github.com/groundx/capstonefuzz/raw/master/corpus/corpus-libFuzzer-capstone_fuzz_disasmnext-latest.zip
-        - unzip -q corpus-libFuzzer-capstone_fuzz_disasmnext-latest.zip -d suite/fuzz
-        # TODO remove built in cmocka compile and use system cmocka (including brewfile) once xenial is default
-        - git clone https://git.cryptomilk.org/projects/cmocka.git suite/cstest/cmocka
-        - chmod +x suite/cstest/build_cstest.sh
+  - wget https://github.com/groundx/capstonefuzz/raw/master/corpus/corpus-libFuzzer-capstone_fuzz_disasmnext-latest.zip
+  - unzip -q corpus-libFuzzer-capstone_fuzz_disasmnext-latest.zip -d suite/fuzz
```
Comment (aquynh) [inline .travis.yml]: this PR is for master branch, but above lines are about next branch? or do you actually want to update the next branch?

Comment (stephengroat) [reply .travis.yml]: I based this off of `master`, didn't know about the `next` branch

**Comment by XVilka** [comment]
Can you please update it and aim to the "next" branch instead? Nobody uses master these days as it's too old.

**Comment by pranith** [comment]
Can you please create a new PR on [libcapstone](https://github.com/libcapstone/libcapstone)?

**Pull request closed without merging**